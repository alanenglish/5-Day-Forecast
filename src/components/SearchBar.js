import React from 'react';
import PropTypes from 'prop-types';

class SearchBar extends React.Component {
  static propTypes = {
    onSubmit: PropTypes.func.isRequired
  }

  state = {
    woeId: ''
  }

  onSubmit = (e) => {
    e.preventDefault();

    if (this.state.woeId !== '') {
      this.props.onSubmit(this.state.woeId);
    }

    this.setState({ woeId: '' });
  }

  render() {
    return (
      <form className="search-bar" onSubmit={(e) => { this.onSubmit(e); }} >
        <span className="fa fa-search search-bar__icon" />
        <input
          className="search-bar__input"
          type="number"
          placeholder="provide a WOE ID and press enter to display forecast for that location..."
          value={this.state.woeId}
          onChange={(e) => { this.setState({ woeId: e.target.value }); }}
          autoFocus
        />
      </form>
    );
  }
}

export default SearchBar;
