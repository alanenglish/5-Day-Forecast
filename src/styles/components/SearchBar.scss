.search-bar {
  align-items: center;
  background: #5597ED;
  box-shadow: 0 4px 4px -4px #777;
  display: flex;
  height: 10vh;
  justify-content: center;
  width: 100vw;

  .search-bar__input {
    border: 2px solid white;
    border-radius: 5rem;
    height: 35%;
    margin-left: 5px;
    padding-left: 10px;
    margin-right: 40px;
    width: 40%;
    &::placeholder {
      font-style: italic;
    }
    &:focus {
      outline: none;
    }
  }

  .search-bar__icon {
    color: white;
    font-size: 120%;
  }

}

@media only screen and (max-width: 768px) {
    /* For mobile phones: */
    .search-bar {
      height: 6vh;

    .search-bar__input {
      height: 60%;
      width: 65%;
      margin-right: 20px;
      &::placeholder {
        font-size: 80%;
      }
    }
  }
}
