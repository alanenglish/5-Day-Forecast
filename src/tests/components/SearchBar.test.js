import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from '../../components/SearchBar';
/* eslint-env jest */

let onSubmitSpy;

beforeEach(() => {
  onSubmitSpy = jest.fn();
});

test('should render SearchBar correctly', () => {
  const wrapper = shallow(<SearchBar onSubmit={onSubmitSpy} />);
  expect(wrapper).toMatchSnapshot();
});

test('should set WOE ID on input change', () => {
  const value = 44418;
  const wrapper = shallow(<SearchBar onSubmit={onSubmitSpy} />);

  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });

  expect(wrapper.state('woeId')).toBe(value);
});

test('should call onSubmit prop for valid for submission', () => {
  const wrapper = shallow(<SearchBar onSubmit={onSubmitSpy} />);
  const value = 44418;

  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });

  wrapper.find('form').simulate('submit', {
    preventDefault: () => { }
  });

  expect(onSubmitSpy).toHaveBeenLastCalledWith(value);
  expect(wrapper.state('woeId')).toBe('');
});
