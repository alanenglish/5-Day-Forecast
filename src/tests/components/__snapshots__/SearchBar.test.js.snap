// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchBar correctly 1`] = `
<form
  className="search-bar"
  onSubmit={[Function]}
>
  <span
    className="fa fa-search search-bar__icon"
  />
  <input
    autoFocus={true}
    className="search-bar__input"
    onChange={[Function]}
    placeholder="provide a WOE ID and press enter to display forecast for that location..."
    type="number"
    value=""
  />
</form>
`;
